Assets/MixedRealityToolkit.Services/InputSimulation/KeyBinding.cs

```diff
@@ -0,0 +1,255 @@
+        /// </summary>
+        public enum MouseButton : int
+        {
+            Left = 0,
+            Right = 1,
+            Middle = 2,
+            Button3 = 3,
+            Button4 = 4,
+            Button5 = 5,
+            Button6 = 6,
+            Button7 = 7,
+        }
+
+        // Array of names to use for a combined enum selection.
+        internal static readonly string[] AllCodeNames;
+        // Maps (KeyType, code) combination onto the contiguous index used for enums.
+        // Value can be used as index in AllCodeNames array.
+        internal static readonly Dictionary<Tuple<KeyType, int>, int> KeyBindingToEnumMap;
+        // Maps enum index to a KeyBinding, for assignment after selecting an enum value.
+        internal static readonly Dictionary<int, KeyBinding> EnumToKeyBindingMap;
+
+        // Static constructor to initialize static fields
+        static KeyBinding()
+        {
+            KeyCode[] KeyCodeValues = (KeyCode[])Enum.GetValues(typeof(KeyCode));
+            string[] KeyCodeNames = Enum.GetNames(typeof(KeyCode));
+            MouseButton[] MouseButtonValues = (MouseButton[])Enum.GetValues(typeof(MouseButton));
+            string[] MouseButtonNames = Enum.GetNames(typeof(MouseButton));
+
+            // Build maps for converting between int enum value and KeyBinding values
+            {
+                KeyBindingToEnumMap = new Dictionary<Tuple<KeyType, int>, int>();
+                EnumToKeyBindingMap = new Dictionary<int, KeyBinding>();
+                List<string> names = new List<string>();
+
+                int index = 0;
+                Action<KeyType, int> AddEnumValue = (bindingType, code) =>
+                {
+                    var kb = new KeyBinding() { bindingType=bindingType, code=code };
+                    names.Add(kb.ToString());
+                    EnumToKeyBindingMap[index] = kb;
+                    KeyBindingToEnumMap[Tuple.Create(bindingType, code)] = index;
+
+                    ++index;
+                };
+
+                AddEnumValue(KeyType.None, 0);
+
+                foreach (MouseButton mb in MouseButtonValues)
+                {
+                    AddEnumValue(KeyType.Mouse, (int)mb);
+                }
+
+                foreach (KeyCode kc in KeyCodeValues)
+                {
+                    AddEnumValue(KeyType.Key, (int)kc);
+                }
+
+                AllCodeNames = names.ToArray();
+            }
+        }
+
+        [SerializeField]
+        private KeyType bindingType;
+        public KeyType BindingType => bindingType;
+        [SerializeField]
+        private int code;
+        internal int Code => code;
+
+        /// <inheritdoc />
+        public override string ToString()
+        {
+            string s = "";
+
+            s += bindingType.ToString();
+
+            switch (bindingType)
+            {
+                case KeyType.Key:
+                    s += ": " + ((KeyCode)code).ToString();
```

Comment:
>((KeyCode)code).ToString(); [](start = 32, length = 27)

Another thing to consider in the future:

[https://docs.microsoft.com/en-us/dotnet/api/system.enum.getname?view=netframework-4.8](https://docs.microsoft.com/en-us/dotnet/api/system.enum.getname?view=netframework-4.8)

For this, probably not worth changing to that link.